src/python/busca_binaria_nao_recursiva.py

```diff
@@ -0,0 +1,66 @@
+"""
+Busca binária (não recursiva)
+
+Objetivo: Implementar o algoritmo de busca binária de forma não recursiva
+          (evitando da propria função ter de ser chamar diversas vezes)
+          com o intuito de ser mais simples de ser explicado ao ir cortando o array
+
+Por: Samuel Sampaio @samukasmk [20230306]
+"""
+
+
+def busca_binaria_nao_recursiva(array_de_elementos, elemento_para_buscar):
+    # repete esse laço condicional enquanto ainda existirem elementos no array
+    while len(array_de_elementos) > 0:
+        # obtem o elemento do meio do array
+        indice_do_meio = len(array_de_elementos) // 2
+        elemento_do_meio = array_de_elementos[indice_do_meio]
+
+        if elemento_para_buscar == elemento_do_meio:
+            # elemento localizado retornando verdadeiro
+            return True
+
+        elif elemento_para_buscar > elemento_do_meio:
+            # corta o array atual apenas obtendo os elementos da direita
+            array_de_elementos = array_de_elementos[indice_do_meio + 1:]
+
+        elif elemento_para_buscar < elemento_do_meio:
+            # corta o array atual apenas obtendo os elementos da esquerda
+            array_de_elementos = array_de_elementos[:indice_do_meio]
+
+    # se chegar ao fim do arrays cortados e o elemento não for encontrado retorna falso
+    return False
```

Comment:
Seria mais interessante retornar o índice (ou -1 se não for encontrado) ao invés de retornar um booleano.